snapcraft/internal/lifecycle/_runner.py

```diff
@@ -277,11 +285,18 @@ def _run_step(self, *, step: steps.Step, part, progress, hint=''):
 
         part = _replace_in_part(part)
 
+    def _run_step(self, *, step: steps.Step, part, progress, hint=''):
+        self._prepare_to_run(step=step, part=part)
```

Comment:
maybe `_prepare_to_run_step`?

Reply:
Went with `_prepare_step` so we have `_{prepare,run,complete}_step`.